volume/local/local_unix.go

```diff
@@ -26,21 +28,24 @@ var (
 		"type":   {}, // specify the filesystem type for mount, e.g. nfs
 		"o":      {}, // generic mount options
 		"device": {}, // device to mount from
+		"size":   {}, // quota size limit
 	}
-	mandatoryOpts = map[string]struct{}{
-		"device": {},
-		"type":   {},
+	mandatoryOpts = map[string][]string{
```

Comment:
Prior to this, as soon as any option on the local_unix driver was set, all options in this map(or rather, set) had to be provided.
This is not desirable for the size option, since it can very well stand on its own.
So the new logic changes this to a map of string-arrays, where the array elements indicate which other options are required if the option which is they key is set.
That way all prior mandatory constraints are still maintained, while size can stand on its own without requiring any other options to be set.